gulpfile.js

```diff
@@ -0,0 +1,35 @@
+const gulp = require('gulp');
+const gutil = require('gulp-util');
+const typescript = require('gulp-typescript');
+const sourcemaps = require('gulp-sourcemaps');
+const tsProject = typescript.createProject('tsconfig.json');
+
+const OPTIONS = {
+    header: true,
+    files: {
+        typescript: [ 'lib/**/*.ts' ],
+        declarations: [ 'lib/**/*.d.ts' ]
+    },
+    base: 'lib'
+}
+
+// Compile the TS sources
+gulp.task('typescript', () => {
+    gulp.src(OPTIONS.files.typescript)
+        .pipe(sourcemaps.init())
+        .pipe(tsProject()).on('error', gutil.log)
+        .pipe(sourcemaps.write('./', {
+			includeContent: false,
+            sourceRoot: '../lib'
```

Comment:
I've had to alter slightly how the gulpfile deals with hand-made declaration files to copy them across.

It now uses options from the `tsconfig.json`.